src/ray/stats/metric.h

```diff
@@ -129,6 +129,9 @@ class Metric {
   std::vector<opencensus::tags::TagKey> tag_keys_;
   std::unique_ptr<opencensus::stats::Measure<double>> measure_;
 
+  // For making sure thread-safe to all of metric registrations.
+  static absl::Mutex mutex_;
```

Comment:
Can you rename it to registration_mutex_?

Reply:
> Can you rename it to registration_mutex_?

Nice catch.